astropy/io/fits/column.py

```diff
@@ -1794,11 +1794,7 @@ def __sub__(self, other):
         if not isinstance(other, (list, tuple)):
             other = [other]
         _other = [_get_index(self.names, key) for key in other]
-        indx = list(range(len(self)))
-        for x in _other:
-            indx.remove(x)
-        tmp = [self[i] for i in indx]
-        return ColDefs(tmp)
+        return ColDefs([self[i] for i in range(len(self)) if i not in _other])
```

Comment:
As far as I can tell `ColDefs` is not an iterable, so it is not possible to use `enumerate(self)` here.

Reply:
A quick trial seems to suggest ducktyping is in good form for enumerating: just having `__getitem__` seems to be enough:
```
class A:
   def __getitem__(self, i):
        if i < 4:
            return "abc"[i]
        raise IndexError
 
for i, a in enumerate(A()): print(i, a)
0 a
1 b
2 c
```

Reply:
The code here is not covered by tests, so I don't feel very comfortable editing it more than I already have.